src/error.ts

```diff
@@ -131,15 +135,28 @@ export class MongoError extends Error {
   cause?: Error; // depending on the node version, this may or may not exist on the base
 
   constructor(message: string | Error) {
+    super(MongoError.buildErrorMessage(message));
     if (message instanceof Error) {
-      super(message.message);
       this.cause = message;
-    } else {
-      super(message);
     }
+
     this[kErrorLabels] = new Set();
   }
 
+  /** @internal */
+  private static buildErrorMessage(e: Error | string): string {
+    if (typeof e === 'string') {
+      return e;
+    }
+    if (isAggregateError(e) && e.message.length === 0) {
```

Comment:
```suggestion
    if (error.message.length === 0 && 'errors' in error && Array.isArray(error.errors)) {
```

Maybe we don't need the type assertion any more? b/c you need to make the argument `unknown` vs staying in this function means you've already narrowed to `Error`, less things to check

Reply:
the type assertion provides type safety on the `.errors` property inside the if-block, and it provides better clarify.  I updated the type assertion to only assert on Error types